src/app/features/psbt-signer/hooks/use-inputs-outputs-totals.tsx

```diff
@@ -0,0 +1,61 @@
+import { useMemo } from 'react';
+
+import { createMoney } from '@shared/models/money.model';
+
+import { sumNumbers } from '@app/common/math/helpers';
+
+import { PsbtInput } from './use-parsed-inputs';
+import { PsbtOutput } from './use-parsed-outputs';
+
+function getAddressInputsTotal(address: string, inputs: PsbtInput[]) {
+  return createMoney(
+    sumNumbers(inputs.filter(input => input.address === address).map(input => input.value)),
+    'BTC'
+  );
+}
+
+function getAddressOutputsTotal(address: string, outputs: PsbtOutput[]) {
+  return createMoney(
+    sumNumbers(
+      outputs.filter(output => output.address === address).map(output => Number(output.value))
+    ),
+    'BTC'
+  );
+}
+
+interface UseInputsOutputsTotals {
+  bitcoinAddressNativeSegwit: string;
+  bitcoinAddressTaproot: string;
+  parsedInputs: PsbtInput[];
+  parsedOutputs: PsbtOutput[];
+}
+export function useInputsOutputsTotals({
+  bitcoinAddressNativeSegwit,
+  bitcoinAddressTaproot,
+  parsedInputs,
+  parsedOutputs,
+}: UseInputsOutputsTotals) {
+  const addressNativeSegwitInputTotal = useMemo(
+    () => getAddressInputsTotal(bitcoinAddressNativeSegwit, parsedInputs),
+    [bitcoinAddressNativeSegwit, parsedInputs]
+  );
+  const addressTaprootInputTotal = useMemo(
+    () => getAddressInputsTotal(bitcoinAddressTaproot, parsedInputs),
+    [bitcoinAddressTaproot, parsedInputs]
+  );
+  const addressNativeSegwitOutputTotal = useMemo(
+    () => getAddressOutputsTotal(bitcoinAddressNativeSegwit, parsedOutputs),
+    [bitcoinAddressNativeSegwit, parsedOutputs]
+  );
+  const addressTaprootOutputTotal = useMemo(
+    () => getAddressOutputsTotal(bitcoinAddressTaproot, parsedOutputs),
+    [bitcoinAddressTaproot, parsedOutputs]
+  );
+
+  return {
+    addressNativeSegwitInputTotal,
+    addressTaprootInputTotal,
+    addressNativeSegwitOutputTotal,
+    addressTaprootOutputTotal,
+  };
+}
```

Comment:
```suggestion
export function useInputsOutputsTotals({
  bitcoinAddressNativeSegwit,
  bitcoinAddressTaproot,
  parsedInputs,
  parsedOutputs,
}: UseInputsOutputsTotals) {
  return useMemo(() => ({
    addressNativeSegwitInputTotal: getAddressInputsTotal(bitcoinAddressNativeSegwit, parsedInputs),
    addressTaprootInputTotal: getAddressInputsTotal(bitcoinAddressTaproot, parsedInputs),
    addressNativeSegwitOutputTotal: getAddressOutputsTotal(bitcoinAddressNativeSegwit, parsedOutputs),
    addressTaprootOutputTotal: getAddressOutputsTotal(bitcoinAddressTaproot, parsedOutputs),
  }), [bitcoinAddressNativeSegwit, bitcoinAddressTaproot, parsedInputs, parsedOutputs]);
}
```

A single use memo more readable imo